axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetDerogatoryLineServiceImpl.java

```diff
@@ -338,14 +338,14 @@ public void remove(FixedAssetDerogatoryLine line) {
   }
 
   @Override
-  public void filterListByStatus(
-      List<FixedAssetDerogatoryLine> fixedAssetDerogatoryLineList, int status) {
+  public void filterListByDate(
+      List<FixedAssetDerogatoryLine> fixedAssetDerogatoryLineList, LocalDate date) {
     List<FixedAssetDerogatoryLine> derogatoryLinesToRemove = new ArrayList<>();
     if (fixedAssetDerogatoryLineList != null) {
       fixedAssetDerogatoryLineList.stream()
-          .filter(line -> line.getStatusSelect() == status)
+          .filter(line -> line.getDepreciationDate().isAfter(date))
```

Comment:
I write it here but it works for the whole PR but `depreciationDate` is not required in `FixedAssetDerogatoryLine` so can you check that it's never an issue ?